posthog/hogql/database/s3_table.py

```diff
@@ -0,0 +1,29 @@
+from typing import Dict
+
+from posthog.hogql.database.models import TableFunction, DatabaseField
+from posthog.hogql.escape_sql import escape_hogql_identifier
+
+
+class S3Table(TableFunction):
```

Comment:
I called this `ExternalTable`, as we'll support GCP/Azure and potentially others. Don't feel very strongly about this.